docs/machine.rst

```diff
@@ -0,0 +1,75 @@
+Machine
+*******
+
+.. py:module:: machine
+
+The machine module contains specific functions related to the micro:bit 
+hardware. Most functions in this module allow to achieve direct and 
+unrestricted access to and control of hardware blocks on a system (like CPU, 
+timers, buses, etc.). Used incorrectly, this can lead to malfunction, lockups, 
+crashes of your board, and in extreme cases, hardware damage.
+
+.. note::
+    All callbacks used by functions and class methods of the ``machine`` module
+    should be considered as executing in an interrupt context.
+
+Functions
+=========
+
+.. method:: machine.unique_id()
+
+    Returns a byte string with a unique identifier of a board/SoC. It will vary
```

Comment:
Because we haven't really introduced the meaning of SoC before (System on Chip), we should simplify the wording here and simply say "identifier of a board", and also do the same in the next sentence.